### PR TITLE
update AppInfo type with new allowedAuthMethods field

### PR DESCRIPTION
**Pull request by flanagankp**
### Description

As part of the new response from the Passage service, the full list of enabled authentication methods will be included in the app info endpoint. This PR updates the type information for the appInfo response and marks the old fields as deprecated.

### CODE REVIEW

**Comment by flanagankp** [comment]
Note integration tests will fail until backend work is done

File: Sources/Passage/PassageModels.swift [hunk 1]
```diff
@@ -177,12 +180,49 @@ public struct AppInfo: Codable, Equatable {
     }
     
     /// Which fallback method is set in the Passage Application when Passkeys are not available
+    /// @available(*, deprecated, message: "Check the allowedAuthMethods property for the full list of supported authentication methods and their configurations.")
```
Comment (rickycpadilla) [inline Sources/Passage/PassageModels.swift:183]: We'll need to remove the "/// " from this line before we merge in :)

File: Sources/Passage/PassageModels.swift [hunk 2]
```diff
@@ -168,6 +170,7 @@ public struct AppInfo: Codable, Equatable {
         case requireIdentifierVerification = "require_identifier_verification"
         case sessionTimeoutLength = "session_timeout_length"
         case userMetadataSchema = "user_metadata_schema"
+        case allowedAuthMethods = "allowed_auth_methods"
```
[the remark this answers is not in the document]
Comment (jenn-macfarlane) [reply Sources/Passage/PassageModels.swift:173]: `allowed_auth_methods` is the old key, just wanted to double check, I might not be reading this correctly!

[the remark this answers is not in the document]
Comment (flanagankp) [reply Sources/Passage/PassageModels.swift]: You're right, for some reason I had the field as `allowed_auth_methods` in my head, even though I got it right over on the passage-js/elements side

**Review by rickycpadilla — APPROVED** [review]
This looks great. I ran the ui test (which is failing in GitHub Actions) locally against this branch, and it passes fine.

**Pull request merged**